pyquil/api/_engagement_manager.py

```diff
@@ -78,7 +94,12 @@ def get_engagement(
                     request = CreateEngagementRequest(
                         quantum_processor_id=quantum_processor_id, endpoint_id=endpoint_id or UNSET
                     )
-                    self._cached_engagements[key] = create_engagement(client=client, json_body=request).parsed
+                    response = create_engagement(client=client, json_body=request)
+                    try:
+                        self._cached_engagements[key] = response.parsed
```

Comment:
The exception doesn't come from `create_engagement(...)`?

Reply:
Strange, but true.